src/plugins/validation/2and3/semantic-validators/responses.js

```diff
@@ -10,18 +10,93 @@ module.exports.validate = function({ jsSpec, isOAS3 }, config) {
   result.warning = [];
 
   config = config.responses;
-
   walk(jsSpec, [], function(obj, path) {
     const contentsOfResponsesObject = path[path.length - 1] === 'responses';
     const isRef = !!obj.$ref;
-
     if (contentsOfResponsesObject && !isRef) {
       each(obj, (response, responseKey) => {
         if (isOAS3) {
           each(response.content, (mediaType, mediaTypeKey) => {
-            const hasInlineSchema = mediaType.schema && !mediaType.schema.$ref;
-            if (
-              hasInlineSchema &&
+            if (mediaType && mediaType.schema && mediaType.schema.oneOf) {
+              for (let i = 0; i < mediaType.schema.oneOf.length; i++) {
+                const hasInlineSchema =
+                  mediaType.schema &&
+                  mediaType.schema.oneOf &&
+                  !Object.keys(mediaType.schema.oneOf[i]).includes('$ref');
+                if (hasInlineSchema) {
+                  const checkStatus = config.inline_response_schema;
+                  if (checkStatus !== 'off') {
+                    result[checkStatus].push({
+                      path: [
+                        ...path,
+                        responseKey,
+                        'content',
+                        mediaTypeKey,
+                        'schema',
+                        'oneOf'
```

Comment:
`i` should be included in the path, otherwise there is no way to separate between schemas within the compose schema.